pkg/cloudprovider/backup_cache.go

```diff
@@ -41,15 +41,17 @@ type backupCache struct {
 	// This doesn't really need to be a map right now, but if we ever move to supporting multiple
 	// buckets, this will be ready for it.
 	buckets map[string]*backupCacheBucket
+	logger  *logrus.Logger
 }
 
 var _ BackupGetter = &backupCache{}
```

Comment:
nit: I see these assertions as tests, so maybe they should move to a _test.go file. I also have a a belief that 90% of these assertions are unnecessary as tests will fail if X doesn't implement Y in the general case, and in the specific case that you want to test that X implements an Y, it should be a test case with a type assertion rather than a compilation failure.